source/dub/project.d

```diff
@@ -1405,10 +1405,21 @@ private string[] processVarsWithGlob(Project, Package)(string var, in Project pr
 	}
 	if (i == res.length) //no globbing found in the path
 		return [res];
-	import std.path : globMatch;
 	import std.file : dirEntries, SpanMode;
-	return dirEntries(res[0 .. sepIdx], SpanMode.depth)
-		.map!(de => de.name)
+	import std.path : buildNormalizedPath, globMatch, isAbsolute, relativePath;
+	auto cwd = gsettings.toolWorkingDirectory.toNativeString;
```

Comment:
dirEntries always returns paths relative to the queried directory, what this code does is:

1) if path is not absolute, prepend the toolWorkingDirectory (in regular DUB exe usage this is just the cwd)
2) iterate directory
3) for each item, if path wasn't absolute in step 1, make each item relative to toolWorkingDirectory

so the paths are still always all relative to the cwd.

What is a behavior change is that I used buildNormalizedPath here though, which normalizes also the parts inside the paths, potentially changing a prefix length one might expect. I can change that to just buildPath, but shorter paths are usually better for the compiler invocations, especially on windows, where there is a path limit, but also for logging to the user in verbose mode + some parts of DUB deduplicate based on normalized paths (which is also a reason why I opened this PR, because with changed cwd it didn't do a correct job when compiling)